kano_updater/commands/install.py

```diff
@@ -74,29 +74,29 @@ def install(progress=None, gui=True):
         logger.warn(msg)
         progress.abort(msg)
         return False
-    elif status.state != UpdaterStatus.UPDATES_DOWNLOADED:
-        logger.debug("Updates weren't downloaded, running download first.")
-        progress.split(
-            Phase(
-                'download',
-                _("Downloading updates"),
-                40,
-                is_main=True
-            ),
-            Phase(
-                'install',
-                _("Installing updates"),
-                60,
-                is_main=True
-            ),
-        )
 
-        progress.start('download')
-        if not download(progress):
-            logger.error("Downloading updates failed, cannot update.")
-            return False
+    logger.debug("Downloading any new updates that might be available.")
+    progress.split(
+        Phase(
+            'download',
+            _("Downloading updates"),
+            40,
+            is_main=True
+        ),
+        Phase(
+            'install',
+            _("Installing updates"),
+            60,
```

Comment:
You might want this 60/40 split based on whether some updates were already downloaded because if they were then it is likely that the download isn't going to take as long so maybe it should be a 80/20 split?

Reply:
So this actually makes sure that was downloaded already is the latest thing by running through the `download` phase again (and consequently the `check`). So I can't know it's been downloaded before check/download-ing :)

Reply:
Yeah, that's true, but we do know if it has tried to do a download before because the status will be `UPDATES_DOWNLOADED` so you could set the division based on whether it is that or it is `NO_UPDATES`.